packages/serverpod/lib/src/database/database_connection.dart

```diff
@@ -295,26 +295,23 @@ class DatabaseConnection {
   }) async {
     var startTime = DateTime.now();
 
-    var data = row.toJsonForDatabase();
+    var selectedColumns = columns ?? row.table.columns;
+    Map data = row.allToJson();
 
-    columns?.forEach((column) {
-      if (!data.containsKey(column.columnName)) {
-        throw Exception(
-          'Column ${column.columnName} does not exist in table row provided.',
-        );
-      }
-    });
-
-    int? id = data['id'];
-
-    Iterable<String> cols = columns?.map((c) => c.columnName) ?? data.keys;
+    for (var column in selectedColumns) {
+      assert(data.containsKey(column.columnName));
```

Comment:
Question: Didn't we want this to be an Exception? Otherwise the user will get an sql error in production which are pretty badly formatted unfortunately and don't help much :(

Reply:
Same comment applies for insert.

Reply:
I thought we said assert? 😄 The rationale here is that this should basically never be able to happen in production if you run the code at least once in development mode. Implementing a clientside dynamic model is potentially possible but not a common pattern or solution I think.